package/MDAnalysis/topology/RDKitParser.py

```diff
@@ -0,0 +1,289 @@
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+RDKit topology parser
+=====================
+
+Converts an `RDKit <https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Mol>`_ 
+:class:`rdkit.Chem.rdchem.Mol` into a :class:`MDAnalysis.core.Topology`.
+
+
+Example
+-------
+
+TODO
+
+
+See Also
+--------
+:mod:`MDAnalysis.coordinates.RDKit`
+
+
+Classes
+-------
+
+.. autoclass:: RDKitParser
+   :members:
+   :inherited-members:
+
+"""
+from __future__ import absolute_import, division
+
+import logging
+import warnings
+import numpy as np
+
+from .base import TopologyReaderBase, change_squash
+from . import guessers
+from ..core.topologyattrs import (
+    Atomids,
+    Atomnames,
+    Atomtypes,
+    Elements,
+    Masses,
+    Charges,
+    Bonds,
+    Resids,
+    Resnums,
+    Resnames,
+    Segids,
+    AltLocs,
+    ChainIDs,
+    Occupancies,
+    Tempfactors,
+)
+from ..core.topology import Topology
+
+logger = logging.getLogger("MDAnalysis.topology.RDKitParser")
+
+
+class RDKitParser(TopologyReaderBase):
+    """
+    For RDKit structures
+
+    Create the following Attributes:
+     - Atomids
+     - Atomnames
+     - Elements
+     - Masses
+     - Bonds
+     - Resids
+     - Resnums
+     - Segids
+
+    Guesses the following:
+     - Atomtypes
+
+    Depending on RDKit's input, the following Attributes might be present:
+     - Charges
+     - Resnames
+     - AltLocs
+     - ChainIDs
+     - Occupancies
+     - Tempfactors
+     
+    """
+    format = 'RDKIT'
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Parser read object *thing*?
+
+        .. versionadded:: 1.0.0
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:  # if no rdkit, probably not rdkit
+            return False
+        else:
+            return isinstance(thing, Chem.Mol)
+
+    def parse(self, **kwargs):
+        """Parse RDKit into Topology
+
+        Returns
+        -------
+        MDAnalysis Topology object
+        """
+
+        mol = self.filename
+
+        # Atoms
+        names = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        occupancies = []
+        tempfactors = []
+
+        # check if multiple charges present
+        atom = mol.GetAtomWithIdx(0)
+        if atom.HasProp('_GasteigerCharge') and (
```

Comment:
Might be easier to read if we can move this closer to where we feed `Charges`?

Reply:
I put it outside of the loop because I didn't want to check if both charges are present for every atom. Should I put it after the loop on atoms then ?

Reply:
No I think it's fine, my bad here, I somehow hadn't taken the looping into account.